dotCMS/src/test/java/com/dotcms/system/event/local/LocalSystemEventsAPITest.java

```diff
@@ -212,6 +215,103 @@ public void orphanSubscriberTest() throws DotDataException {
         Assert.assertTrue(isCalled.get());
     } // orphanSubscriberTest.
 
+    /**
+     * Given scenario: We're testing that even when no subscribers are added the asyncNotify callback gets called
+     * Expected results: We expect that the callback gets called.
+     */
+    @Test
+    public void Test_Non_Blocking_Notification_On_Event_Consumed_With_No_Subscribers() {
+        final LocalSystemEventsAPI localSystemEventsAPI = APILocator.getLocalSystemEventsAPI();
+        final AtomicBoolean isOnCompleteCalled = new AtomicBoolean(false);
+        final String message = "Async notify Event.";
+        localSystemEventsAPI.asyncNotify(new TestEventType1(message), event -> {
+            Assert.assertTrue(event instanceof TestEventType1);
+            final TestEventType1 eventType1 = (TestEventType1)event;
+            Assert.assertEquals(eventType1.getMsg(), message);
+            isOnCompleteCalled.set(true);
+        });
+        DateUtil.sleep(2000);
+        Assert.assertTrue(isOnCompleteCalled.get());
+    }
+
+    /**
+     * Given scenario: We're testing that even when subscribers are added  the asyncNotify callback gets called just once.
+     * Expected results: We expect that the callback gets called exactly once.
+     */
+    @Test
+    public void Test_Non_Blocking_Notification_On_Event_Consumed_With_Subscribers() {
+        final String message = "Async notify Event 2.";
+        final LocalSystemEventsAPI localSystemEventsAPI = APILocator.getLocalSystemEventsAPI();
+        localSystemEventsAPI.subscribe(new TestDelegateSubscriber(){
+            @Override
+            public void notify(final TestEventType2 event) {
+                Assert.assertEquals(event.getMsg(), message);
+            }
+        });
+        final AtomicInteger callsCount = new AtomicInteger(0);
 
-
-} // E:O:F:LocalSystemEventsAPITest.
+        localSystemEventsAPI.asyncNotify(new TestEventType2(message), event -> {
+            Assert.assertTrue(event instanceof TestEventType2);
+            final TestEventType2 eventType2 = (TestEventType2)event;
+            Assert.assertEquals(eventType2.getMsg(), message);
+            callsCount.incrementAndGet();
+        });
+        DateUtil.sleep(2000);
+        Assert.assertEquals(callsCount.get(), 1);
+    }
+
+    /**
+     * Given scenario: We're testing that an event that implements KeyFilterable and is broadcast only reaches certain audience.
+     * We creating a few subscribers each one of the provides a key. The Event must reach only those which have the same id.
+     * Expected results: Only the subscribers with a matching id will be able to consume the event;
+     */
+    @Test
+    public void Test_Send_Event_For_Limited_Audience() {
+
+        final Map<String, AtomicInteger> callsCounts = ImmutableMap
+                .of("subscriber1", new AtomicInteger(0), "subscriber2", new AtomicInteger(0),
+                        "subscriber3", new AtomicInteger(0));
```

Comment:
Codacy found an issue: [The String literal "subscriber3" appears 7 times in this file; the first occurrence is on line 273](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5856383)